File: Ahorn/entities/BarrelBomber.jl
module BarrelBomber

using ..Ahorn, Maple

@mapdef Entity "MaggyHelper/BarrelBomber" BarrelBomber(x::Integer, y::Integer, detectionRange::Integer=80, explosionRadius::Integer=100, fuseTime::Number=1.5, health::Integer=1)

const placements = Ahorn.PlacementDict(
    "normal" => Ahorn.EntityPlacement(BarrelBomber)
)

function Ahorn.selection(entity::BarrelBomber)
    return Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::BarrelBomber, room::Maple.Room)
    Ahorn.drawRectangle(ctx, Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16), (0.5, 0.5, 0.5, 0.5), (0.0, 0.0, 0.0, 1.0))
end

end

File: framework/extracted/Loenn/entities/barrelBomber.lua
local utils = require("utils")
local drawableSprite = require("structs.drawable_sprite")

local barrelBomber = {}

barrelBomber.name = "MaggyHelper/BarrelBomber"
barrelBomber.depth = -50
barrelBomber.fieldInformation = {
    health = {
        minimumValue = 1
    },
    detectionRange = {
        minimumValue = 0
    },
    explosionRadius = {
        minimumValue = 0
    },
    fuseTime = {
        minimumValue = 0
    }
}
barrelBomber.placements = {
    {
        name = "normal",
        data = {
            health = 1,
            detectionRange = 80,
            explosionRadius = 100,
            fuseTime = 1.5
        }
    }
}

function barrelBomber.sprite(room, entity)
    local sprite = drawableSprite.fromTexture("characters/barrel_bomber/hidden", entity)
    sprite:setPosition(entity.x, entity.y)
    sprite:setJustification(0.5, 1.0)
    return sprite
end

function barrelBomber.selection(room, entity)
    return utils.rectangle(entity.x - 12, entity.y - 32, 24, 32)
end

return barrelBomber
